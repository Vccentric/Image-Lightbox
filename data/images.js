import img1 from "../public/images/img_mountains_wide.jpg";
import img2 from "../public/images/img_nature_wide.jpg";
import img3 from "../public/images/img_snow_wide.jpg";

export const images = [
  { src: img1, alt: "Spring Mountains" },
  { src: img2, alt: "Fall Mountains" },
  { src: img3, alt: "Winter Mountains" },
];

export default images;
