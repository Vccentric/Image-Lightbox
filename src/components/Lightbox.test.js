import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import React from "react";
import Lighbox from "./Lightbox";
import images from "../../data/images";

// setup variables
const toggleOpen = true;
const currentImage = images[0];
const close = jest.fn();
const prev = jest.fn();
const next = jest.fn();

test("renders correctly - snapshot", () => {
  const tree = renderer
    .create(
      <Lighbox
        toggleOpen={toggleOpen}
        currentImage={currentImage}
        close={close}
        prev={prev}
        next={next}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("renders lightbox toggle open", () => {
  const { container } = render(
    <Lighbox
      toggleOpen={toggleOpen}
      currentImage={currentImage}
      close={close}
      prev={prev}
      next={next}
    />
  );
  expect(container.querySelector("#lightbox")?.className).toEqual("");
  expect(screen.getByText("Spring Mountains")).toBeInTheDocument();
});

test("renders lightbox with different image than default", () => {
  render(
    <Lighbox
      toggleOpen={toggleOpen}
      currentImage={images[1]}
      close={close}
      prev={prev}
      next={next}
    />
  );
  expect(screen.getByText("Fall Mountains")).toBeInTheDocument();
});
