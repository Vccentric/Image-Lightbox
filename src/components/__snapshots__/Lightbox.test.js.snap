// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly - snapshot 1`] = `
<div
  className=""
  id="lightbox"
>
  <div
    id="overlay"
  />
  <div
    id="modal"
  >
    <button
      className="close"
      onClick={[MockFunction]}
    >
      X
    </button>
    <div
      className="image-container"
    >
      <div
        className="full-image"
      >
        <img
          alt="Spring Mountains"
          className="selected-image"
          onClick={[MockFunction]}
          src=""
        />
      </div>
      <span
        className="caption"
      >
        Spring Mountains
      </span>
    </div>
    <div
      className="controls-container"
    >
      <button
        className="prev"
        onClick={[MockFunction]}
      >
        &lt;&lt; PREV
      </button>
      <button
        className="next"
        onClick={[MockFunction]}
      >
        NEXT &gt;&gt;
      </button>
    </div>
  </div>
</div>
`;
